core/unit_tests.c

```diff
@@ -60,6 +61,210 @@ unit_test_atomic_ops(void);
 void
 unit_test_jit_fragment_tree(void);
 
+#ifdef __AVX__
```

Comment:
Better to use a DR-specified define, driven by the check_avx_processor_and_compiler_support. Have it set a cmake var and define it in configure.cmake.h. We'd have one for avx512 too. Then we avoid compiler-specific assumptions here and can port this to other platforms more easily.

Reply:
Actually this I kind of disagree, because the __AVX__ etc. variables are the standard that mainstream compilers set. So I wanted to make this part independent of the rest, it "just works". If the test was compiled with AVX it is defined, if not, it is not. This is platform independent. The __AVX__ tells you that this file was compiled with support for AVX. Leave it? Re-open if necessary.

Reply:
If both __AVX__ and __AVX512__ are set by all our compilers (including VS) then I could live with this though it is not what I would vote for.

My logic is that we have a check (check_avx_processor_and_compiler_support) that is more than just the compiler: we check whether the processor and OS  have support.  It seems weird to then throw that out the window and go and compile avx code based on a completely different conditional that is just the compiler.  I would expect all our avx code to be triggered by the same top-level check.  If some platform does not enable avx yet the compiler has it on by default, this code as written will crash, despite the check_avx_* doing the right thing.

Note that we avoid gcc's `__i386__` or VS's `_WIN64` in favor of defines driven by our top-level config.

Reply:
> If both AVX and AVX512 are set by all our compilers (including VS) then I could live with this though it is not what I would vote for

Yes, VS does, too.

To your point, the test crashing then is actually a good thing! If the compiler has it on by default (this won't happen any time soon.. considering that processes don't shrink much anymore), but if it does (that would be a cross compile), then DynamoRIO will be compiled with AVX as well and crash on the same machine.